### PR TITLE
Improvements to logging

### PR DESCRIPTION
**Pull request by jenshnielsen**
* Avoid calling getLogger on each logging call. This avoids 2 or 3 unnecessary function calls for each log call
* Don't set the level on each individual logger to a hardcoded constant when getting the logger. This makes it hard to be able to consume the loggers since every time getLogger is called the level would be overwritten. Rather leave the level unset and rely on the level of the parent logger. 
* To enable the above ensure that the level of the "plottr" logger is set to match the level of the handler attached to it when setting up a handler (both for stream an qt widget handler)
* Add a commandline argument to enable the user to toggle the console output level for easier debugging.

### CODE REVIEW

File: plottr/apps/inspectr.py [hunk 1]
```diff
@@ -669,5 +666,8 @@ def script() -> None:
     parser = argparse.ArgumentParser(description='inspectr -- sifting through qcodes data.')
     parser.add_argument('--dbpath', help='path to qcodes .db file',
                         default=None)
+    parser.add_argument("--console-log-level",
+                        choices=("ERROR", "WARNING", "INFO", "DEBUG"),
+                        default="WARNING")
     args = parser.parse_args()
-    main(args.dbpath)
+    main(args.dbpath, args.console_log_level)
```
Comment (astafan8) [inline plottr/apps/inspectr.py:673]: shouldn't the value of the argument be converted to int that corresponds to the level?

**Review by astafan8 — APPROVED** [review]

File: plottr/apps/inspectr.py (hunk 1 above)
Comment (jenshnielsen) [reply plottr/apps/inspectr.py:673]: It works like this too:

From the docs of [setLevel](https://docs.python.org/3/library/logging.html#logging.Logger.setLevel)

> Changed in version 3.2: The level parameter now accepts a string representation of the level such as ‘INFO’ as an alternative to the integer constants such as INFO. Note, however, that levels are internally stored as integers, and methods such as e.g. [getEffectiveLevel()](https://docs.python.org/3/library/logging.html#logging.Logger.getEffectiveLevel) and [isEnabledFor()](https://docs.python.org/3/library/logging.html#logging.Logger.isEnabledFor) will return/expect to be passed integers.


And also confirmed manually

```
In [1]: import logging
In [2]: logger = logging.getLogger()
In [3]: logger.setLevel("WARNING")
In [4]: logger.level
Out[4]: 30
In [5]: logger.setLevel("INFO")
In [6]: logger.level
Out[6]: 20
```

but happy to change if you find it more clear the other way around

Comment (jenshnielsen) [reply plottr/apps/inspectr.py:673]: but the types should be updated then to match this

Comment (astafan8) [reply plottr/apps/inspectr.py:673]: oh, cool, didn't know!

it's fine either way :)

**Pull request merged**